composer/models/huggingface.py

```diff
@@ -51,30 +52,20 @@ def __init__(self,
                  metrics: Optional[List[Metric]] = None) -> None:
         try:
             import transformers
+            del transformers  # unused
         except ImportError as e:
             raise MissingConditionalImportError(extra_deps_group='nlp', conda_package='transformers') from e
```

Comment:
What is the point of this if we are not using the transformers import?

Reply:
Although we don't use the `transformers` import, you won't be able to use this class if `transformers` is not installed in your environment, because you have to pass in a model/tokenizer with a type from `transformers`. This error just makes it explicit, rather than crashing in some other less clear way. I'm not super attached to this pattern, but it follows a pattern from some other places in our code (e.g. https://github.com/mosaicml/composer/blob/658c4852880e53c0c26c3a5974d06a1532b3e6f1/composer/loggers/wandb_logger.py#L66-L73)

Reply:
maybe in this case its better to just delete if/until this class actually needs access to the package

Reply:
Ok, removed.